dialogflow/session_entity_type_management.rb

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "securerandom"
+
+def list_session_entity_types project_id:, session_id:
+  # [START dialogflow_list_session_entity_types]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "Existing Session ID"
+  
+  require "google/cloud/dialogflow"
+
+  session_entity_types_client = Google::Cloud::Dialogflow::SessionEntityTypes.new
+  session_path = session_entity_types_client.class.session_path project_id, session_id
+
+  session_entity_types = session_entity_types_client.list_session_entity_types session_path
+
+  puts "SessionEntityTypes for session #{session_path}:\n"
+  session_entity_types.each do |session_entity_type|
+    entity_values = session_entity_type.entities.map { |e| e.value }
+    puts "=" * 20
+    puts "SessionEntityType name: #{session_entity_type.name}"
+    puts "Numer of entities:      #{session_entity_type.entities.size}"
+    puts "SessionEntityType entity values: #{entity_values}"
+  end
+  # [END dialogflow_list_session_entity_types]
+end
+
+
+def create_session_entity_type project_id:, session_id:,
+                               entity_type_display_name:,
+                               entity_values:
+  # [START dialogflow_create_session_entity_type]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "Existing Session ID"
+  # entity_type_display_name = "Existing Entity Type Display Name"
+  # entity_values = ["entity1", "entity2"]
+  
+  require "google/cloud/dialogflow"
+
+  session_entity_types_client = Google::Cloud::Dialogflow::SessionEntityTypes.new
+  session_path = session_entity_types_client.class.session_path project_id, session_id
+  session_entity_type_name = session_entity_types_client.class.session_entity_type_path project_id, session_id, entity_type_display_name
+
+  # Here we use the entity value as the only synonym.
+  entities = entity_values.map { |entity_value| { value: entity_value, synonyms: [entity_value] } }
+
+  session_entity_type = { 
+    name:                 session_entity_type_name,
+    entities:             entities,
+    # the modes are either ENTITY_OVERRIDE_MODE_OVERRIDE or ENTITY_OVERRIDE_MODE_SUPPLEMENT
+    entity_override_mode: :ENTITY_OVERRIDE_MODE_OVERRIDE
+  }
+
+  response = session_entity_types_client.create_session_entity_type session_path, session_entity_type
+
+  puts "SessionEntityType created: #{response}"
```

Comment:
What is the result of outputing the `response` object?

Vote: don't print out `response`

Reply:
Agreed - for Ruby this prints something that's not useful.